### PR TITLE
Interface & Visualization

### PR DESCRIPTION
**Pull request by im-prakher**

### CODE REVIEW

**Comment by KristinaRiemer** [comment]
So I was trying to follow along with [the interface README](https://github.com/PecanProject/BETYdb-YABA/blob/50088c7badbb66692ca19d4b4eb2ac6fb58479e5/interface/README.md). When I ran `yarn start`, it returned this error: `Command failed with exit code 127`. After some Googling, apparently just `yarn` needs to be run first, and then `yarn start` worked. So that should be added to the README. 

This is how the first page looks for me: 
Screen Shot 2020-06-26 at 8 23 23 AM

File: app/Meta.py
Comment (Chris-Schnaufer) [inline app/Meta.py:146]: Is BETYdb YABA only going to be used to load data from the Phoenix area? If not, a means of allowing the user to specify the GMT offset should be added

File: app/Meta.py
Comment (Chris-Schnaufer) [inline app/Meta.py]: I feel the term "some text" should be replaced with "No notes". I feel that this is clearer

File: app/Meta.py
Comment (Chris-Schnaufer) [inline app/Meta.py:136]: This is an unsafe way of assigning polygons to plot names in that it assumes the same order between plot names and geometries, which is only possible, not probable. Polygons should be assigned based upon plot name or a method that guarantees correct assignment. I believe there's an issue that refers to this problem

File: app/Meta.py
Comment (Chris-Schnaufer) [inline app/Meta.py]: `os.remove(file_name)` should remain here so that it's cleaned up when a problem occurs. Checking if the file exists before trying to remove it would prevent a missing file exception from being raised
```
if os.path.exists(file_name):
    os.remove(file_name)
```

File: client/client.py
Comment (Chris-Schnaufer) [inline client/client.py:21]: An open CORS is very dangerous from a security standpoint. Can this be limited to where the Cross Origin content comes from?

File: interface/public/f-tw.svg
Comment (Chris-Schnaufer) [inline interface/public/f-tw.svg:4]: Remove extra character at the end of this line

File: interface/public/tw.svg
Comment (Chris-Schnaufer) [inline interface/public/tw.svg:4]: Remove extra character at end of this line and in other files that contain the extra character

File: interface/src/App.test.js
Comment (Chris-Schnaufer) [inline interface/src/App.test.js:1]: This testing file appears to be left over from the site generating software. Can this be updated to check for something that's actually on the App page?

File: interface/src/components/navbar.js
Comment (Chris-Schnaufer) [inline interface/src/components/navbar.js]: Better alt text may be "PEcAn Project" for text readers and visual consistency, or "Pecan Project" for text readers

File: interface/src/components/navbar.js
Comment (Chris-Schnaufer) [inline interface/src/components/navbar.js]: Better alt text may be "PEcAn Project on GitHub"

File: interface/src/components/navbar.js
Comment (Chris-Schnaufer) [inline interface/src/components/navbar.js:10]: Why isn't this display style part of the "Link" element CSS? The bar may change its elements spacing when the user clicks it & it may be better to start out with correct spacing.

File: interface/src/components/upload1.js
Comment (Chris-Schnaufer) [inline interface/src/components/upload1.js]:
```suggestion
        alert(`You have not uploaded your ${str} files`);
```

File: visualization/pg_joins.js
Comment (Chris-Schnaufer) [inline visualization/pg_joins.js]: This should be "No notes". See above

File: visualization/pg_joins.js
Comment (Chris-Schnaufer) [inline visualization/pg_joins.js]: Use a more descriptive name here instead of "getTable1". Perhaps insertCultivarSites?

File: visualization/pg_joins.js
Comment (Chris-Schnaufer) [inline visualization/pg_joins.js]: Use a more descriptive name than "getTable2" here

File: visualization/pg_joins.js
Comment (Chris-Schnaufer) [inline visualization/pg_joins.js]: Use "No notes" here as well

File: visualization/pg_joins.js
Comment (Chris-Schnaufer) [inline visualization/pg_joins.js]: Use "No notes" here and everywhere else soilnotes defaults to ~~"some text"~~

File: visualization/pg_joins.js
Comment (Chris-Schnaufer) [inline visualization/pg_joins.js:99]: Why the ROLLBACK here and elsewhere in this file? I'm not sure I understand the purpose of this file if it's kept.

**Review by Chris-Schnaufer — COMMENTED** [review]
Can you comment where you're using generated files vs. ones that you created and/or modified. I think this would help reviewers.
Adding comments to files as to the intent of the file would help people understand what they're about. Especially for example, upload1.js and upload2.js.
Python documentation strings are missing in python code

File: client/client.py
Comment (im-prakher) [reply client/client.py:21]: Yeah!! Will do this in future, before containerizing the interface.

File: interface/src/components/navbar.js
Comment (im-prakher) [reply interface/src/components/navbar.js]: Right! Will add this in next commit.

File: interface/src/components/navbar.js
Comment (im-prakher) [reply interface/src/components/navbar.js]: Right! Will add this in next commit

File: visualization/pg_joins.js
Comment (im-prakher) [reply visualization/pg_joins.js]: Right! Will change this here and everywhere else in next commit

File: visualization/pg_joins.js
Comment (im-prakher) [reply visualization/pg_joins.js]: Thanks! for pointing out, will change it in a future commit.

File: visualization/pg_joins.js
Comment (im-prakher) [reply visualization/pg_joins.js]: Thanks! for pointing out, will change it in a future commit.

File: interface/public/tw.svg
Comment (im-prakher) [reply interface/public/tw.svg:4]: Will fix these typos in next commit

File: interface/src/components/navbar.js
Comment (im-prakher) [reply interface/src/components/navbar.js:10]: Thanks! that's a good point, will surely look into it.

File: visualization/pg_joins.js
Comment (im-prakher) [reply visualization/pg_joins.js:99]: As here we just need the data after performing joins on some tables, for only visualization purposes and every other operation performed on the database before taking table joins (like `INSERT`)  must not be applied to the database. For insertion purposes, we have YABA API which can perform data ingestion after a user has validated the data. Therefore, a 
`ROLLBACK` operation is performed at last after taking joins.
You can also take a look at this issue: #41

File: app/Meta.py
Comment (Chris-Schnaufer) [inline app/Meta.py]: This line of code adds another dimension to the loaded point. This causes problems when the points already have a Z value when loaded from the shapefile (point has 3 dimensions); after this line the point will have 4 dimensions and cause a failure

File: app/Meta.py
Comment (dlebauer) [inline app/Meta.py]:
```suggestion
            data['soilnotes'].fillna("", inplace = True)      
```

convention in the database is empty string

File: app/Meta.py
Comment (dlebauer) [inline app/Meta.py:185]:
```suggestion
        os.remove(shx_file_target)                
        os.remove(file_name)        
```

Comment (dlebauer) [reply app/Meta.py:185]: per Chris' comment https://github.com/PecanProject/BETYdb-YABA/pull/50/files#r447854871

<--

File: client/client.py
Comment (dlebauer) [reply client/client.py:21]: @im-prakher if you are going to do this later, can you create a new issue on closing CORS before merging the PR?

File: interface/public/tw.svg
Comment (dlebauer) [reply interface/public/tw.svg:4]: @Chris-Schnaufer are you talking about the red circle w/ a line in it? These are github indicators that the file doesn't end in a newline.

File: visualization/pg_joins.js
Comment (dlebauer) [inline visualization/pg_joins.js]:
```suggestion
    sites.push("")
```

File: visualization/pg_joins.js
Comment (dlebauer) [inline visualization/pg_joins.js]:
```suggestion
    sites.push("")
```

Comment (dlebauer) [inline visualization/pg_joins.js]:
```suggestion
    sites.push("")
```

Comment (dlebauer) [inline visualization/pg_joins.js]:
```suggestion
    sites.push("")
```

File: visualization/pg_joins.js
Comment (dlebauer) [inline visualization/pg_joins.js]:
```suggestion
                  site.push("")
```

Comment (dlebauer) [inline visualization/pg_joins.js]:
```suggestion
                  site.push("")
```

**Review by dlebauer — CHANGES_REQUESTED** [review]
* @im-prakher try `" "` instead of no notes?
* @Chris-Schnaufer could you provide more explicit instructions or go ahead and implement your suggestions below for .shp file and time zone handling? It isn't clear how challenging this would be but I think especially the shapefile handling could become a rabbit hole for someone not as familiar w/ GIS as you.

File: interface/src/components/navbar.js
Comment (dlebauer) [inline interface/src/components/navbar.js]:
```suggestion
                <a href="http://pecanproject.org/" rel="noopener noreferrer" target="_blank"><img src="/pecan.jpeg" alt="PEcAn Project logo" className="nav-img"/></a>
```

File: interface/README.md
Comment (dlebauer) [inline interface/README.md:9]: Use code blocks for indicating code, e.g. 

```sh
yarn
```

Comment (dlebauer) [reply interface/README.md:9]: * link to code for installing yarn https://classic.yarnpkg.com/en/docs/install
* describe OS requirements
* tell user to `cd interface`

File: app/Meta.py
Comment (saurabh1969) [reply app/Meta.py:146]: Time_zone is filled with "America/Phoenix" when there is no value to that column (fillNA() function finds the NA values). @im-prakher You can replace it with some standard default value for timezone.

File: app/Meta.py
Comment (saurabh1969) [reply app/Meta.py]: **os.remove(file_name)**  should be there.Otherwise file contents will be appended with existing ones when that endpoint is called next time.

**Comment by KristinaRiemer** [comment]
@dlebauer or @Chris-Schnaufer would you be able to confirm your review so that this PR can be merged?

File: app/Meta.py
Comment (Chris-Schnaufer) [inline app/Meta.py]:
```suggestion
                if len(pt) < 3:
                    pt=pt+(115,)
```

Comment (Chris-Schnaufer) [reply app/Meta.py]: Another comment on this is that every point that doesn't have a Z value will be assigned an elevation of 115 meters. IMO it would be better to default to zero or some other value indicating that it's probably not correct (unless you're at sea level)

File: app/Meta.py
Comment (Chris-Schnaufer) [reply app/Meta.py:146]: @saurabh1969 @dlebauer Would GMT +0:00 be a good default?

File: visualization/pg_joins.js
Comment (Chris-Schnaufer) [reply visualization/pg_joins.js:99]: Thank you for explaining. Can you please comment the code to that effect so that it's clearer in the future?

**Comment by im-prakher** [comment]
This PR is now complete in itself, please have a final review and then merge it.
@Chris-Schnaufer @KristinaRiemer @dlebauer @saurabh1969

**Comment by im-prakher** [comment]
This PR is now complete in itself, please have a final review and then merge it.
@Chris-Schnaufer @KristinaRiemer @dlebauer @saurabh1969

**Comment by im-prakher** [comment]
Due to a lot of lines of code in this PR, it might be challenging to review the work done in recent commits so to make this easier, I have opened issues(regarding the recent commits) that are implemented in this PR.
- [Visualize the shapefiles](https://github.com/PecanProject/BETYdb-YABA/issues/56) 
- [Added a Success Page](https://github.com/PecanProject/BETYdb-YABA/issues/57)
- [Added an Error Page](https://github.com/PecanProject/BETYdb-YABA/issues/58)
- [Added requests file to hit YABA app client routes](https://github.com/PecanProject/BETYdb-YABA/issues/59)
- [Created a validation/review page](https://github.com/PecanProject/BETYdb-YABA/issues/41)

File: app/Meta.py
Comment (dlebauer) [reply app/Meta.py:146]: if time zone isn't required, can leave it out (or insert empty string `''`). 
Otherwise could look it up based on lat/lon. [Create an issue]

File: client/client.py
Comment (dlebauer) [reply client/client.py:21]: will be handled in https://github.com/PecanProject/BETYdb-YABA/issues/53

File: interface/src/App.test.js
Comment (dlebauer) [reply interface/src/App.test.js:1]: addressed - file deleted and other tests added in https://github.com/PecanProject/BETYdb-YABA/pull/60

File: visualization/pg_joins.js
Comment (dlebauer) [reply visualization/pg_joins.js:99]: comment added in line 55 of this file https://github.com/PecanProject/BETYdb-YABA/pull/50/files#diff-d8b2fc15cf3be3bac93b61f112240f96R55

File: app/Meta.py
Comment (dlebauer) [reply app/Meta.py:136]: fixed in line 139

**Review by dlebauer — APPROVED** [review]

**Pull request merged**
